addons/im_livechat/static/src/models/thread/thread.js

```diff
@@ -53,21 +55,52 @@ registerClassPatchModel('mail.thread', 'im_livechat/static/src/models/thread/thr
                     )
                 );
                 data2.members.push(['link', partner]);
+                data2.livechatCorrespondent = [['link', partner]];
             } else {
                 const partnerData = this.env.models['mail.partner'].convertData(data.livechat_visitor);
                 data2.members.push(['insert', partnerData]);
+                data2.livechatCorrespondent = [['insert', partnerData]];
             }
         }
         return data2;
     },
 });
 
+registerFieldPatchModel('mail.thread', 'im_livechat/static/src/models/thread/thread.js', {
+    /**
+     * Override correspondent field to compute it properly in the case of
+     * livechat. This must be equivalent to `livechatCorrespondent` field.
+     */
+    correspondent: many2one('mail.partner', {
+        dependencies: ['livechatCorrespondent'],
+    }),
+    /**
+     * Reference to partner being the actual correspondent of chat.
+     *
+     * This field is useful because livechat are not necessarily 2 users
+     * chats, and the correspondent could be either anonymous or logged in.
+     */
+    livechatCorrespondent: many2one('mail.partner'),
+});
+
 registerInstancePatchModel('mail.thread', 'im_livechat/static/src/models/thread/thread.js', {
 
     //----------------------------------------------------------------------
     // Private
     //----------------------------------------------------------------------
 
+    /**
+     * @override
+     */
+    _computeCorrespondent() {
```

Comment:
Would prefer not introducing "crap", but fix flow that assume correspondent is set.

Reply:
For a moment, I agreed with you, but now I see it's not "crap" code. Sure the notion of "correspondent" was first introduced with DM chat and this is common concern with livechat which share the idea, at the time, that these are 2-person users.

Livechat can have many operators with invite... so there were some wrong ideas... but the idea of "correspondent" still holds in my opinion, in that livechat is still identified by the public livechat member. This is important to keep this, notably for some features like typing notification (in case of anonymous public member) and thread icon with chat avatar (a feature to [re-]introduce btw.).

Reply:
Ok but then no `livechatCorrespondent` field.
Just write directly on `correspondent`, and ensure compute does not overwrite the given value when in the case of livechat.